### DISCUSSION TITLE
Problem calling Ghostcript to convert from PDF to image

### DISCUSSION THREAD

**Discussion by wolverine4277** [symfony/symfony#57513] [Q&A]
Hi, I'm new to Synfony and I'm having a problem trying to convert a PDF file to an image file using Ghostcript.
The error is as follows:

> The command "gs -dSAFER -dBATCH -dNOPAUSE -sDEVICE=png16m -dTextAlphaBits=4 -dGraphicsAlphaBits=4 -r144 "-sOutputFile=C:\Windows\TEMP\pdf\25fffc8bdf3fee86\"^%"03d.png" "C:\inetpub\wwwroot\pdf/files/qr2.pdf"" failed. Exit Code: 1(General error)

The part of the code where it fails is:

```
        $this->cacheDir = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'pdftoimage' . DIRECTORY_SEPARATOR . bin2hex((new Secure())->generate());
        $this->filesystem->mkdir($this->cacheDir);
        (new Process([
            'gs',
            '-dSAFER',
            '-dBATCH',
            '-dNOPAUSE',
            '-sDEVICE=png16m',
            '-dTextAlphaBits=4',
            '-dGraphicsAlphaBits=4',
            '-r' . $this->resolution,
            '-sOutputFile=' . $this->cacheDir . DIRECTORY_SEPARATOR . '%03d.png',
            $this->pdfFile])
        )->mustRun();

```
I clarify that running the command from the terminal the image file is generated correctly:

gs -dSAFER -dBATCH -dNOPAUSE -sDEVICE=png16m -dTextAlphaBits=4 -dGraphicsAlphaBits=4 -r144 "-sOutputFile=C:\Windows\TEMP\pdf\25fffc8bdf3fee86\"^%"03d.png" "C:\inetpub\wwwroot\pdf/files/qr2.pdf""

I am using IIS with PHP 8.3.8 NTS (I also tried with TS and got the same error), the user has write permission on the folders, and Ghostcript accessible from the PHP process.
Regards.

Sebastián

**Comment by Trismegiste** [comment]
I have a question : the Process instance is running in a Command or in a Controller ?

Because, in a Controller, the user most likely is different from the terminal.
Apart from the fact user rights could be different, environment variables too.

**Reply by wolverine4277** [reply]
I'm testing it in two ways: in a simple class (which does not inherit from anything) and directly in a script without any class, same result. I don't know if that answers your question. 
Regards.

**Reply by Trismegiste** [reply]
It's probably a memory limit somewhere in the stack but I can't help you for that, I didn't use windows since 2011, sorry
